### ISSUE TITLE
DA Prices seems to have broken. 

### ISSUE THREAD

**Issue by PeteGilbert98** [EnergieID/entsoe-py#346]
Hi @fboerman ,

I've noticed that as of today, the function getting the da_prices from entsoe seems to have broken. My assumption is that the format sent from the entsoe api must have changed.

![image](https://github.com/user-attachments/assets/360eae06-cfe4-4bbc-9e8a-337c87d98832)

As you can see, all prices now seem to be the same. 

Here's what I think might be causing the issue:

On line 36, I would have expected the _extract_timeseries generator to produce multiple soup objects for the different days that I have queried. Instead only one object is returned.

![image](https://github.com/user-attachments/assets/6d2d3474-1507-47ff-a360-72b16a1f843b)

This means, on line 88, the keys in the dictionary (on line 83) are overwritten, as the keys are non unique per day. 

![image](https://github.com/user-attachments/assets/a9688bf3-3b3f-4033-9047-5971e6536fac)

on line 92, the datetime index appears to be correct, covering the whole timespan of the (buffered) query.

When the (seemingly correct) index is combined with the data. Things start to break.

**Comment by LuisTellezSirocco** [comment]
Hi, I'm not here to answer you about the problem, but to tell you that you have leaked your API Key in one of the images...

**Comment by PeteGilbert98** [comment]
> Hi, I'm not here to answer you about the problem, but to tell you that you have leaked your API Key in one of the images...

Thanks, I changed token

**Comment by borg42** [comment]
I have the same problem. It worked fine yesterday, but now i have the same problem as described by @PeteGilbert98. I think this means that Entso-E must have changed something in the XML, right?

**Comment by fboerman** [comment]
Dear all, thank you for the fast report. I see my dashboard also having the same behaviour now. I am currently on holiday without access to my computer but will be back soon. In the meantime a pull request with a fix is always welcome!

⁣Get BlueMail for Android ​

**Comment by binboupan** [comment]
As of today I am getting entsoe.exceptions.NoMatchingDataError, it worked fine for years so something has definitely changed.

**Comment by fboerman** [comment]
Please put this in a proper pull request and I'll look at it after my holiday thanks.

⁣Get BlueMail for Android ​

**Comment by PeteGilbert98** [comment]
@fboerman will do this eve. thanks

**Comment by GeneralCP** [comment]
for anyone looking for a quick fix: https://github.com/GeneralCP/entsoe2

**Comment by binboupan** [comment]
day ahead prices are also broken; all of the prices are the same.

**Comment by JaniKallankari** [comment]
Day ahead prices seems to be broken. Does this change in Entsoe platform https://transparency.entsoe.eu/news/widget?id=66f5203d792e84032cbb9b71 have something to do with this?

**Comment by borg42** [comment]
> Day ahead prices seems to be broken. Does this change in Entsoe platform https://transparency.entsoe.eu/news/widget?id=66f5203d792e84032cbb9b71 have something to do with this?

Yes, this is exactly the problem. The day ahead prices use "variable sized blocks" now:
![vsb](https://github.com/user-attachments/assets/e07d1622-96ab-4faa-a44a-228e9cbe3d78)

while entsoe-py expects every position to be present.

**Comment by Roeland54** [comment]
There is another change I have noticed. If I request prices for today and for tomorrow (belgium) the prices for today using the 60min resolution as usual. But the prices of tomorrow are using the 15min resolution.

![image](https://github.com/user-attachments/assets/aca9f18f-f70d-4a97-89c2-a234addc30a9)

**Comment by JaniKallankari** [comment]
I made a parser for the new data type. Only tested with day a head prices so be carefully. Response all TimeSeries are combined to one pandas.Series with possibly (not likely) none equal time steps. Fell free to use this code you find this code usefully.

`import xml.etree.ElementTree as ET
    def parse_timeseries(self, xml_text, value_key='price.amount', to_float=True):    
        resolution_map = {
            'PT60M': pd.Timedelta(60, 'min'),
            'P1Y'  : pd.Timedelta(365,'day'),
            'PT15M': pd.Timedelta(15, 'min'),
            'PT30M': pd.Timedelta(30, 'min'),
            'P1D'  : pd.Timedelta(1,  'day'),
            'P7D'  : pd.Timedelta(7,  'day'),
            'P1M'  : pd.Timedelta(30, 'day'),
        }

        time_stamps = []
        values      = []
        xml_text = re.sub(' xmlns="[^"]+"', '', xml_text, count=1) #Remove namespace
        root        = ET.fromstring(xml_text)
        for time_serie in root.findall('TimeSeries'):
            #curve_type = time_serie.find('curveType').text
            for period in time_serie.findall('Period'):
                start_time = pd.Timestamp(period.find('timeInterval').find('start').text)
                resolution = resolution_map[period.find('resolution').text]
                for point in period.findall('Point'):
                    position = float(point.find('position').text)-1
                    time_stamps.append(start_time+position*resolution)
                    if to_float:
                        values.append(float(point.find(value_key).text))
                    else:
                        values.append(point.find(value_key).text)
        return pd.Series(data=values,index=time_stamps)
`

**Comment by miikasda** [comment]
The above fix by [JaniKallankari](https://github.com/JaniKallankari) works for day ahead prices. Here is a small guide to include it as a a hotfix to entsoe-py if others need it as well:

Edit entsoe/parsers.py in Python's site-packages and add following:
```
# Hotfix for dayahead prices, see GitHub issue below for more information
# https://github.com/EnergieID/entsoe-py/issues/346
import xml.etree.ElementTree as ET
import re
def parse_timeseries(xml_text, value_key='price.amount', to_float=True):    
    resolution_map = {
        'PT60M': pd.Timedelta(60, 'min'),
        'P1Y'  : pd.Timedelta(365,'day'),
        'PT15M': pd.Timedelta(15, 'min'),
        'PT30M': pd.Timedelta(30, 'min'),
        'P1D'  : pd.Timedelta(1,  'day'),
        'P7D'  : pd.Timedelta(7,  'day'),
        'P1M'  : pd.Timedelta(30, 'day'),
    }
    time_stamps = []
    values      = []
    xml_text = re.sub(' xmlns="[^"]+"', '', xml_text, count=1) #Remove namespace
    root        = ET.fromstring(xml_text)
    for time_serie in root.findall('TimeSeries'):
        #curve_type = time_serie.find('curveType').text
        for period in time_serie.findall('Period'):
            start_time = pd.Timestamp(period.find('timeInterval').find('start').text)
            resolution = resolution_map[period.find('resolution').text]
            for point in period.findall('Point'):
                position = float(point.find('position').text)-1
                time_stamps.append(start_time+position*resolution)
                if to_float:
                    values.append(float(point.find(value_key).text))
                else:
                    values.append(point.find(value_key).text)
    return pd.Series(data=values,index=time_stamps)
```

And then change the  entsoe/entsoe.py to use this new parser. First import the new parser function in line [14](https://github.com/EnergieID/entsoe-py/blob/647ae2bb08d31d218914b7a632eb5542825b2911/entsoe/entsoe.py#L14):
```
from .parsers import parse_timeseries, parse_prices, parse_loads, parse_generation, \
    parse_installed_capacity_per_plant, parse_crossborder_flows, \
    parse_unavailabilities, parse_contracted_reserve, parse_imbalance_prices_zip, \
    parse_imbalance_volumes_zip, parse_netpositions, parse_procured_balancing_capacity, \
    parse_water_hydro,parse_aggregated_bids, parse_activated_balancing_energy_prices
```

And then change the query_day_ahead_prices function defined in line [1202](https://github.com/EnergieID/entsoe-py/blob/647ae2bb08d31d218914b7a632eb5542825b2911/entsoe/entsoe.py#L1202) to actually use it:
```
def query_day_ahead_prices(
            self, country_code: Union[Area, str],
            start: pd.Timestamp,
            end: pd.Timestamp,
            resolution: Literal['60min', '30min', '15min'] = '60min') -> pd.Series:
        """
        Parameters
        ----------
        resolution: either 60min for hourly values,
            30min for half-hourly values or 15min for quarterly values, throws error if type >
        country_code : Area|str
        start : pd.Timestamp
        end : pd.Timestamp

        Returns
        -------
        pd.Series
        """
        if resolution not in ['60min', '30min', '15min']:
            raise InvalidParameterError('Please choose either 60min, 30min or 15min')
        area = lookup_area(country_code)
        # we do here extra days at start and end to fix issue 187
        text = super(EntsoePandasClient, self).query_day_ahead_prices(
            country_code=area,
            #start=start-pd.Timedelta(days=1),
            start = start,
            end = end
            #end=end+pd.Timedelta(days=1)
        )
        #series = parse_prices(text)[resolution]
        series = parse_timeseries(text)
        if len(series) == 0:
            raise NoMatchingDataError
        series = series.tz_convert(area.tz)
        series = series.truncate(before=start, after=end)
        # because of the above fix we need to check again if any valid data exists after trun>
        if len(series) == 0:
            raise NoMatchingDataError
        return series
```

**Comment by fboerman** [comment]
@JaniKallankari @miikasda thank you for your suggestions. as a tip, its usually much more readable if you enter such proposals as a pull request instead of a copy in an issue. I am using your code as hint for the fix I am currently writing for this issue. Many thanks to all in this thread for their suggestions!

**Comment by borg42** [comment]
> resolution = resolution_map[period.find('resolution').text]
>             for point in period.findall('Point'):
>                 position = float(point.find('position').text)-1
>                 time_stamps.append(start_time+position*resolution)
>                 if to_float:
>                     values.append(float(point.find(value_key).text))
>                 else:
>                     values.append(point.find(value_key).text)

This only works if the xml only contains one resolution, which is generally not the case. E.g. for DE_LU it usually contains 60min and 15min data. This would need to return a dict with the resolutions with one series per resolution to work with all regions.

**Comment by fboerman** [comment]
many thanks to all for the discussions, I have released a new version, 0.6.9, which fixes this issue. If you still encounter issues on this version please open a new issue!

**Comment by fboerman** [comment]
oh and @Roeland54  this is probably a mistake on Elia side I believe. I have send them an informal nudge hopefully they will fix it soon, I cant fix that in the package
